problems/cipher.js

```diff
@@ -0,0 +1,259 @@
+    let alphaIndex = null;
+
+    while (alphaIndex === null) {
+        alphabet.split('').forEach((alphaChar) => {
+            if (alphaChar === char) {
+                alphaIndex = alphabet.indexOf(alphaChar);
+            }
+        });
+    };
+  return alphaIndex;
+};
+
+/**
+ * The opposite of charToAlphaIndex.
+ * Takes in a character index and an alphabet string. Returns the character at
+ * the given index using the alphabet string. For instance, for the regular alphabet:
+ * index = 0 returns 'a'
+ * index = 25 returns 'z'
+ * index = 26 returns 'a'
+ * We use % to convert the index to its corresponding index when wrapped around the alphabet.
+ * For instance, an index of 40 % 26 = 14, so we use the new index of 14 to find the correct
+ * character.
+ * @param  {int} index    integer representing the index of the character
+ * @param  {str} alphabet an alphabet string to match the index of the character against
+ * @return {str}          a single character at that position in the alphabet
+ */
+const _indexToChar = (index, alphabet) => {
+    // If we're given a negative index, we need to add the alphabet length
+    // to convert it to its positive match
+    if (index < 0) {
+        index = index + alphabet.length;
+    }
+    // We need to use % to make sure the index will fit on the alphabet
+    // e.g., if we're given an index of 40, that doesn't fit on the alphabet.
+    const convertedIndex = index % alphabet.length;
+
+    return alphabet.charAt(convertedIndex);
+};
+
+/**
+ * Validates a given key to ensure that the key is long enough to be able to
+ * properly encode or decode the stringToConvert. If the key is not long enough,
+ * this method will repeat the key characters until the key becomes long enough.
+ * For instance, given a key of 'ab' and a stringToConvert of 'sahar', the key will
+ * return as 'ababa'.
+ * // TO DO: Add more validation for non-alpha keys
+ * @param  {str} key            the key provided by the user
+ * @param  {str} stringToConvert the string we're taking action on to convert later
+ * @return {str}                the validated key
+ */
+const _validateKey = (key, stringToConvert) => {
+    if (key.length != stringToConvert.length) {
+        index = -1;
+        while (key.length != stringToConvert.length) {
+            index += 1;
+            key = key.concat(key.charAt(index));
+        }
+    }
+    return key;
+};
+
+/**
+ * The opposite of decode.
+ * Takes in a string and returns an encoded string. Uses the provided key
+ * to encode each individual character according to the matching key character
+ * (by index). For instance, given a key of 'bbbb': string = 'abcd' returns 'bcde'
+ * @param  {str} stringToEncode user inputted string to encode
+ * @return {str}                encoded string
+ */
+Cipher.prototype.encode = function(strToEncode) {
+    let encodedArray = [];
+    let charStrIndex = -1;
+
+    let key = _validateKey(this.key, strToEncode);
```

Comment:
on the second hand, this does look much cleaner; just calling the function like that. So maybe for readability your approach of coming up with the matching length key might be a better approach. I'll let you decide. Either way we can make the validateKey function little better =)